dali/operators/generic/slice/slice_attr.h

```diff
@@ -246,6 +239,26 @@ class PositionalSliceAttr {
   }
 
  private:
+  template <typename T>
+  span<T> Span(const TensorListView<StorageCPU, T, DynamicDimensions>& v, int sample_idx) {
+    ptrdiff_t vol = volume(v.tensor_shape_span(sample_idx));
+    return span<T>(v.tensor_data(sample_idx), vol);
+  }
+
+  template <typename T, typename Backend>
+  const TensorListView<StorageCPU, T, DynamicDimensions> GetInputArgumentCPU(
+      const workspace_t<Backend>& ws, int idx, TensorList<CPUBackend>& cpu_buffer) {
+    if (ws.template InputIsType<CPUBackend>(idx)) {
+      const auto &arg = ws.template Input<CPUBackend>(idx);
+      return view<T>(arg);
+    } else {
+      const auto &arg = ws.template Input<GPUBackend>(idx);
+      cpu_buffer.set_order(AccessOrder::host());
```

Comment:
I think we should make the `cpu_buffer` pinned, call async copy, and sync on our own. There would be no speed deference from the DALI point of view but we may avoid blocking other cuda calls if the internal driver staging buffer is occupied and cudamemcopy locks the driver until the buffer is free.

Reply:
I was thinking about explicit pinning, too. This buffer should be small, o eve with restricted pinned memory usage we should be able to afford it.